pkg/destroy/aws/shared.go

```diff
@@ -199,27 +198,38 @@ func (o *ClusterUninstaller) cleanSharedRoute53(ctx context.Context, session *se
 
 	switch resourceType {
 	case "hostedzone":
-		return o.cleanSharedHostedZone(ctx, client, id, logger)
+		return o.cleanSharedHostedZone(ctx, session, id, logger)
 	default:
 		logger.Debugf("Nothing to clean for shared %s resource", resourceType)
 		return nil
 	}
 }
 
-func (o *ClusterUninstaller) cleanSharedHostedZone(ctx context.Context, client *route53.Route53, id string, logger logrus.FieldLogger) error {
+func (o *ClusterUninstaller) cleanSharedHostedZone(ctx context.Context, session *session.Session, id string, logger logrus.FieldLogger) error {
+	publicClient := route53.New(session)
+
+	// The private hosted zone (phz) may belong to a different account,
+	// in which case we need a separate client.
+	phzClient := publicClient
```

Comment:
Not gonna lie, this is a bit hard to read/validate with the back-and-forth of the clients.

Reply:
Maybe something like this would be easier to read?

```go
	// The private hosted zone (phz) may belong to a different account,
	// in which case we need a separate client.
        publicZoneClient  := route53.New(session)
	privateZoneClient := route53.New(session)
```

Reply:
I think it does but no need to change it unless you also have to make other changes.

Reply:
I think I am going to have a follow up PR to address this and @mtulio's [feedback](https://github.com/openshift/installer/pull/7161#pullrequestreview-1456036731). I also think I should update the in-repo docs with an example install config